test/integration/acceptance/grant_permission_test.cpp

```diff
@@ -0,0 +1,702 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "acceptance_fixture.hpp"
+#include "builders/protobuf/queries.hpp"
+#include "builders/protobuf/transaction.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+using namespace shared_model::interface;
+
+class GrantPermissionTest : public AcceptanceFixture {
+ public:
+  using TxBuilder = TestUnsignedTransactionBuilder;
+
+  /**
+   * Creates the transaction with the user creation commands
+   * @param perms are the permissions of the user
+   * @return built tx and a hash of its payload
+   */
+  auto makeAccountWithPerms(
+      const std::string &user,
+      const crypto::Keypair &key,
+      const shared_model::interface::RolePermissionSet &perms,
+      const std::string &role) {
+    return createUserWithPerms(user, key.publicKey(), role, perms)
+        .build()
+        .signAndAddSignature(kAdminKeypair)
+        .finish();
+  }
+
+  /**
+   * Creates two accounts with corresponding permission sets
+   * @param perm1 set of permissions for account #1
+   * @param perm2 set of permissions for account #2
+   * @return reference to ITF object with two transactions
+   */
+  auto createTwoAccounts(
+      const shared_model::interface::RolePermissionSet &perm1,
+      const shared_model::interface::RolePermissionSet &perm2) {
+    auto itf = std::make_shared<IntegrationTestFramework>(1);
```

Comment:
I don't think that ITF was expected to be used in this way. Maybe create ITF in test body, and pass it here as a reference? You can also return it as a reference from this method.